specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/stable/2021-04-01/SmartDetectorAlertRulesApi.json

```diff
@@ -0,0 +1,750 @@
+            "$ref": "./examples/SmartDetectorAlertRule_CreateOrUpdate.json"
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "SmartDetectorAlertRules"
+        ],
+        "operationId": "SmartDetectorAlertRules_Patch",
+        "description": "Patch a specific Smart Detector alert rule.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/AlertRuleNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "description": "Parameters supplied to the operation.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/AlertRulePatchObject"
+            }
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/SmartDetectorErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Successful request to patch Smart Detector alert rule.",
+            "schema": {
+              "$ref": "#/definitions/AlertRule"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Patch alert rules": {
+            "$ref": "./examples/SmartDetectorAlertRule_Patch.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "SmartDetectorAlertRules"
+        ],
+        "operationId": "SmartDetectorAlertRules_Delete",
+        "description": "Delete an existing Smart Detector alert rule.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/AlertRuleNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/SmartDetectorErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Successful request to delete a Smart Detector alert rule."
+          },
+          "204": {
+            "description": "The Smart Detector alert rule does not exist. It may have already been deleted."
+          }
+        },
+        "x-ms-examples": {
+          "Delete a Smart Detector alert rule": {
+            "$ref": "./examples/SmartDetectorAlertRule_Delete.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SmartDetectorErrorResponse": {
```

Comment:
>SmartDetectorErrorResponse [](http://example.com/codeflow?start=5&length=26)

Please consider using ErrorResponse definition in the common type.json. [https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json)